docs/src/references/hsm-support.md

```diff
@@ -35,201 +26,164 @@ Additionally, the built-in bridge has to be used.
 
 This feature has the following related configuration options:
 
+```sh command="tedge config list --doc device.cryptoki" title="tedge config list --doc device.cryptoki"
 ```
-     device.cryptoki.enable  Use a Hardware Security Module for authenticating the MQTT connection with the cloud.  When set to true, `key_path` option is ignored as PKCS#11 module is used for signing.
-                             Examples: true, false
-device.cryptoki.module_path  A path to the PKCS#11 module used for interaction with the HSM.
-                             Example: /usr/lib/x86_64-linux-gnu/opensc-pkcs11.so
-        device.cryptoki.pin  Pin value for logging into the HSM. 
-                             Example: 123456
-     device.cryptoki.serial  A serial number of a Personal Identity Verification (PIV) device to be used.  Necessary if two or more modules are connected. 
-                             Example: 123456789
+
+```sh command="tedge config list --doc key_uri" title="tedge config list --doc key_uri"
 ```
 
 ## Setup guide
+<!-- split the guide into a separate page under "Operate Devices" category? -->
+<!-- maybe it would be better to use SoftHSM2 instead of a Yubikey so everyone can follow? -->
+<!-- also would be nice to write a test for this guide -->
 
 The following guide shows how to connect to Cumulocity using a PIV-capable Yubikey. The guide uses a
 Yubikey 5C NFC.
 
-### Setup the Yubikey
+### Part 1: Setup the cryptographic token
 
 1. Make sure the Yubikey is connected.
-1. Install the [ykman CLI](https://docs.yubico.com/software/yubikey/tools/ykman/).
-2. Create a private key on the device. In the guide we'll import the private key, but you can also
-   create one directly on the Yubikey to ensure its contents are never read by the device.
+<!--is it possible to use a generic p11 client to setup the token instead of a yubikey-specific client?-->
+2. Install the [ykman CLI](https://docs.yubico.com/software/yubikey/tools/ykman/).
+3. Create a private key on the device. In the guide we'll import the private key, but you can also
+   create one directly on the Yubikey to ensure its contents are not exposed.
 
-```sh
-ykman piv keys import 9a $(tedge config get device.key_path)
-```
+    ```sh
+    ykman piv keys import 9a $(tedge config get device.key_path)
+    ```
```

Comment:
Since most of these commands are Yubikey specific we might need to reorganize the docs (or maybe pick a different token by default which is more pkcs11 compliant, like Nitrokey HSM 2)

Reply:
The best would probably be to use SoftHSM2, so everyone can follow the instructions.

Reply:
changed in 22324daa